buildSrc/src/main/groovy/com/jakewharton/kotlingen/KotlinGenTask.groovy

```diff
@@ -0,0 +1,282 @@
+      return doc.replace("<em>", "*")
+          .replace("</em>", "*")
+          .replace("<p>", "")
+
+          // JavaParser adds a couple spaces to the beginning of these for some reason
+          .replace("   *", " *")
+
+          // {@code view} -> `view`
+          .replaceAll(/\{@code ($DOC_LINK_REGEX)\}/) { String fullmatch, String codeName -> "`$codeName`" }
+
+          // {@link Foo} -> [Foo]
+          .replaceAll(/\{@link ($DOC_LINK_REGEX)\}/) { String fullmatch, String foo -> "[$foo]" }
+
+          // {@link Foo#bar} -> [Foo.bar]
+          .replaceAll(/\{@link ($DOC_LINK_REGEX)#($DOC_LINK_REGEX)\}/) { String fullmatch, String foo, String bar -> "[$foo.$bar]" }
+
+          // {@linkplain Foo baz} -> [baz][Foo]
+          .replaceAll(/\{@linkplain ($DOC_LINK_REGEX) ($DOC_LINK_REGEX)\}/) { String fullmatch, String foo, String baz -> "[$baz][$foo]" }
+
+          //{@linkplain Foo#bar baz} -> [baz][Foo.bar]
+          .replaceAll(/\{@linkplain ($DOC_LINK_REGEX)#($DOC_LINK_REGEX) ($DOC_LINK_REGEX)\}/) { String fullmatch, String foo, String bar, String baz -> "[$baz][$foo.$bar]" }
+          .trim()
+    }
+
+    /** Generates method level type parameters */
+    private static String typeParams(List<TypeParameter> params) {
+      if (!params || params.isEmpty()) {
+        return null
+      }
+
+      StringBuilder builder = new StringBuilder()
+      builder.append("<")
+      params.each { p -> builder.append("${p.name} : ${p.typeBound[0].name}") }
+      builder.append(">")
+      return builder.toString()
+    }
+
+    /**
+     * Generates parameters in a kotlin-style format
+     *
+     * @param specifyType boolean indicating whether or not to specify the type (i.e. we don't
+     *        need the type when we're passing params into the underlying Java implementation)
+     */
+    private String kParams(boolean specifyType) {
+      StringBuilder builder = new StringBuilder()
+      parameters.each { p -> builder.append("${p.id.name}${specifyType ? ": " + resolveKotlinType(p.type) : ""}") }
+      return builder.toString()
+    }
+
+    /**
+     * Generates the kotlin code for this method
+     *
+     * @param bindingClass name of the RxBinding class this is tied to
+     */
+    public String generate(String bindingClass) {
+      ///////////////
+      // STRUCTURE //
+      ///////////////
+      // Javadoc
+      // public inline fun DrawerLayout.drawerOpen(): Observable<Boolean> = RxDrawerLayout.drawerOpen(this)
+      // <access specifier> inline fun <extendedClass>.<name>(params): <type> = <bindingClass>.name(this, params)
+
+      String fParams = kParams(true)
+      String jParams = kParams(false)
+      String generated =
+              "${comment ? comment : ""}\n" +
+                      "$accessModifier inline fun ${typeParameters ? typeParameters + " " : ""}$extendedClass.${name}($fParams): ${resolveKotlinType(returnType)} = $bindingClass.$name(${jParams ? "this, $jParams" : "this"})"
```

Comment:
Can we break this up? Either into multiple strings with a single combining statement or just as a series of concatenations for each part.